Gemfile

```diff
@@ -14,6 +14,9 @@ gem 'tzinfo-data', platforms: [:mswin, :mswin64]
 # Bundle edge Rails instead: gem 'rails', github: 'rails/rails'
 gem 'rails', '~> 7.2.1'
 
+# Soft delete
+gem 'paranoia'
```

Comment:
The `paranoia` gem github page also recommended to look at `discard` gem

> paranoia has some surprising behaviour (like overriding ActiveRecord's delete and destroy) and is not recommended for new projects. See [discard's README](https://github.com/jhawthorn/discard#why-not-paranoia-or-acts_as_paranoid) for more details.

Have you taken a look at `discard`?

Reply:
Learning the discard.